### PR TITLE
KAN-40-formatted-output-from-sanity-check

### PR DESCRIPTION
**Pull request by wkhu-astro**

### CODE REVIEW

File: museek/plugin/sanity_check_observation_plugin.py [hunk 1]
```diff
@@ -317,3 +320,60 @@ def check_closeness_to_sunrise_sunset(self, data: TimeOrderedData, report_writer
...
+        azimuth_min = np.min(data.azimuth.array)
+        azimuth_max = np.max(data.azimuth.array)
+        dec_min = np.min(data.declination.array)
+        dec_max = np.max(data.declination.array)
+        ra_min = np.min(data.right_ascension.array)
+        ra_max = np.max(data.right_ascension.array)
+        targets = [i for i in data.obs_script_log if 'Observation targets' in i][0]
+        targets = targets.split('Observation')[-1]
+
+        header = "block number  |  Description  |  observation start date/time (UTC)  |  observation duration (minutes) |  scan start date/time  | scan duration (minutes) |  observation start - nearest sunset (minutes)  | nearest sunrise - observation end (minutes)  |  number of dishes used after stragglers are removed  |  elevation check  |  elevation mean |  azimuth range  |  declination range  |  ra range  |  targets observed"
```
Comment (amadeuswi) [inline museek/plugin/sanity_check_observation_plugin.py:375]: lines are too long :)

Comment (wkhu-astro) [reply museek/plugin/sanity_check_observation_plugin.py]: shortened

**Comment by mariogrs** [comment]
Looking at the output:
block number | Description | observation start date/time (UTC) | observation duration (minutes) | scan start date/time | scan duration (minutes) | obs. start - nearest sunset (minutes) | nearest sunrise - obs. end (minutes) | num of dishes used (after stragglers are removed) | elevation check | elevation mean | azimuth range | declination range | ra range | targets observed

1666630286_sdp_l0 | Description = 'UHF HI intensity mapping Field 1 Rising  no.9 offset=50' | 2022-10-24 16:51:54.276804 | 133.6938 | 2022-10-24 17:10:06.592301 | 97.3166 | 0.4682 | 519.2993 | 58 | OK | 41.7674 | 44.0577~66.9067 | -5.2149~6.8293 | -16.2034~19.5719 |  targets are ['PKS 2153-69_u0.8', 'PKS 2153-69', 'PKS 2153-69_d0.8', 'PKS 2153-69_r0.8', 'PKS 2153-69_l0.8', 'scan_azel_with_nd_trigger']

- remove _sdp_l0  from block number
- use separate columns for min and max instead of e.g. 44.0577~66.9067 
- is ra range correct?
- add one column to say 'rise' or 'set'

**Review by mariogrs — APPROVED** [review]
I didn't go through the code in detail but it seems to be running well. Make sure you don't include the stragglers when counting the number of dishes with bad elevation.

**Review by amadeuswi — COMMENTED** [review]
Do the existing unittests for `time_analysis` and `from_log` still run?

**Comment by amadeuswi** [comment]
I am worried that some unittests are broken with this PR

File: museek/antenna_sanity/from_log.py [hunk 2]
```diff
@@ -24,6 +24,6 @@ def straggler_list(self) -> list[str]:
                 for antenna_name in addition:
                     straggler_list.append(antenna_name[1:-1])
 
-                unique_straggler_list = sorted(set(straggler_list))
+        unique_straggler_list = sorted(set(straggler_list))
```
Comment (amadeuswi) [inline museek/antenna_sanity/from_log.py:27]: please check unittests before merging

Comment (wkhu-astro) [reply museek/antenna_sanity/from_log.py:27]: I update the unittests for from_log and time_analysis.  from_log is ok.  time_analysis is ok after a small correction.

File: test/museek/util/test_time_analysis.py [hunk 3]
```diff
@@ -10,9 +10,11 @@ def test_time_difference_to_sunset_sunrise(self):
         obs_end = datetime.fromtimestamp(3600)  # 1 hour later
         start, end, end_diff, start_diff = time_analysis.time_difference_to_sunset_sunrise(obs_start=obs_start, 
                                                                                            obs_end=obs_end)
+
+        print ('start, end, end_diff, start_diff',start, end, end_diff, start_diff)
         self.assertEqual(datetime(1969, 12, 31, 18, 7, 1, 543037), start)
```
Comment (amadeuswi) [inline test/museek/util/test_time_analysis.py:15]: This assert fails when I run it

**Review by amadeuswi — CHANGES_REQUESTED** [review]

File: test/museek/util/test_time_analysis.py [hunk 4]
```diff
@@ -10,9 +10,11 @@ def test_time_difference_to_sunset_sunrise(self):
         obs_end = datetime.fromtimestamp(3600)  # 1 hour later
         start, end, end_diff, start_diff = time_analysis.time_difference_to_sunset_sunrise(obs_start=obs_start, 
                                                                                            obs_end=obs_end)
+
+        print ('start, end, end_diff, start_diff',start, end, end_diff, start_diff)
```
Comment (amadeuswi) [inline test/museek/util/test_time_analysis.py:14]: please remove the print statement here

**Review by amadeuswi — APPROVED** [review]

File: test/museek/util/test_time_analysis.py (hunk 4 above)
Comment (wkhu-astro) [reply test/museek/util/test_time_analysis.py]: removed

**Pull request merged**